VMEncryption/main/handle.py

```diff
@@ -945,14 +1029,21 @@ def handle_encryption(public_settings, encryption_status, disk_util, bek_util, e
             logger.log("An operation already running. Cannot accept an update settings request.")
             hutil.reject_settings()
         are_devices_encrypted, items_to_encrypt = are_required_devices_encrypted(volume_type, encryption_status, disk_util, bek_util, encryption_operation)
-        if not are_devices_encrypted:
-            logger.log('Required devices not encrypted for volume type {0}. Calling update to stamp encryption settings.'.format(volume_type))
-            update_encryption_settings(items_to_encrypt)
-            logger.log('Encryption Settings stamped. Calling enable to encrypt new devices.')
-            enable_encryption()
+        if security_Type==CommonVariables.ConfidentialVM:
+            logger.log('Calling Update Encryption Setting in LUKS2 header.')
+            if extension_parameter.cmk_changed():
+                update_encryption_settings_luks2_header(items_to_encrypt)
+            if not are_devices_encrypted:
+                enable_encryption()
         else:
-            logger.log('Calling Update Encryption Setting.')
-            update_encryption_settings()
+            if not are_devices_encrypted:
+                logger.log('Required devices not encrypted for volume type {0}. Calling update to stamp encryption settings.'.format(volume_type))
+                update_encryption_settings(items_to_encrypt)
+                logger.log('Encryption Settings stamped. Calling enable to encrypt new devices.')
+                enable_encryption()
+            else:
+                logger.log('Calling Update Encryption Setting.')
+                update_encryption_settings()
```

Comment:
Can you help me as to why the if else logic was added here ?

Reply:
else part is for if all devices are encrypted.